src/hotspot/share/jfr/writers/jfrStreamWriterHost.inline.hpp

```diff
@@ -77,7 +79,16 @@ inline void StreamWriterHost<Adapter, AP>::write_bytes(const u1* buf, intptr_t l
   while (len > 0) {
     const unsigned int nBytes = len > INT_MAX ? INT_MAX : (unsigned int)len;
     const ssize_t num_written = os::write(_fd, buf, nBytes);
+
+    if (errno==ENOSPC) {
+        char msg[] = "Failed to write to jfr stream because no space left on device";
+        log_error(jfr, system)("%s", msg);
+        JavaThread* jt = JavaThread::current();
+        ThreadInVMfromNative transition(jt);
+        JfrJavaSupport::abort(JfrJavaSupport::new_string(msg, jt), jt, false);
```

Comment:
Hi again Takuya, I'm sorry, but I should have noticed this earlier:

I now see that the code needs to allocate a Java string oop to conform to the existing abort function signature, which caters to invocations from Java. Then abort() immediately strips out the c-string from the oop. To be correct, also headers for logging/log.hpp and runtime/thread.inline.hpp should need be included.

I believe we can simplify this by updating the abort() signature so that we don't need to drag in those extra dependencies. Please see my following comment where I suggest a way to do this.

Thanks for your patience
Markus

Reply:
Thank you for your valuable comments. I agree with you. I corrected this fix in accordance with your suggestions.